docs/whitepaper/src/appendix.tex

```diff
@@ -0,0 +1,209 @@
+            \State Set ${\mathcal{C}_{\max} \leftarrow \mathcal{C}_{i}}$
+        \Else
+            \lstinline|  // It is assumed that the input chains are the same size|
+            \State Set ${\mathcal{C}_{\max} \leftarrow \mathcal{C}_{i'}}$
+        \EndIf
+        \State \Return ${\mathcal{C}_{\max}}$
+    \end{algorithmic}\label{alg:algorithm-01}
+\end{algorithm}
+
+\begin{algorithm}[ht]
+    \begin{algorithmic}[1]
+        \caption{${\textsf{MaxValidChain}(\mathcal{C}_{\text{loc}}, C_{\text{set}} = \{\mathcal{C}_i\}_{i=1}^N, K_{\text{f}}, S_{\text{g}})}$}
+        \State Set ${\mathcal{C}_{\max} \leftarrow \mathcal{C}_{\text{loc}}}$
+        \lstinline|  // Set local chain as initially maximum valid chain|
+        \For {${i = 1 \dots N}$}
+            \If {$\mathcal{C}_i$ forks from $\mathcal{C}_{\max}$ at most $K_{\text{f}}$ blocks}
+                \If {$|\mathcal{C}_i| > |\mathcal{C}_{\max}|$}
+                    \State Set ${\mathcal{C}_{\max} \leftarrow \mathcal{C}_{i}}$
+                \ElsIf
+                        {$|\mathcal{C}_i| == |\mathcal{C}_{\max}|$}
+                    \State Set ${\mathcal{C}_{\max} \leftarrow \textsf{MaxStakeChain}(\mathcal{C}_{i}, \mathcal{C}_{\max})}$
+
+                \EndIf
+            \Else
+
+                \State Let ${j  \leftarrow \max \{j' \geq 0 \ |\  \mathcal{C}_{\max} \text{ and } \mathcal{C}_{i} \
+                \text{have the same block in } sl_{j'}\}}$
+                \If {$|\mathcal{C}_i[j: j + S_{\text{g}}]| > |\mathcal{C}_{\max}[j: j + S_{\text{g}}]|$}
+                    \State Set ${\mathcal{C}_{\max} \leftarrow \mathcal{C}_{i}}$
+                \ElsIf
+                        {$|\mathcal{C}_i| == |\mathcal{C}_{\max}|$}
+                    \State Set ${\mathcal{C}_{\max} \leftarrow \textsf{MaxStakeChain}(\mathcal{C}_{i}, \mathcal{C}_{\max})}$
+                \EndIf
+            \EndIf
+
+        \EndFor
+        \State \Return ${\mathcal{C}_{\max}}$
+    \end{algorithmic}\label{alg:algorithm-1}
+\end{algorithm}
+
+
+\begin{algorithm}[ht]
+    \begin{algorithmic}[1]
+        \caption{$\textsf{IsValidBlock}(B)$}
+
+        \State Set stakeholders distribution $S, S_{\text{cons}}$ at the end of epoch $e_{j - 1}$, $e_{j - 3}$ in $\mathcal{C}_{\text{loc}}$
+        \State Set verified participants table $V_{\text{ver}}$ derived from epoch $e_{j - 3}$ view in $\mathcal{C}_{\text{loc}}$
+        \State Set $\{T^{\text{cons}, i}\}_{i=1}^K$ as consensus group lottery thresholds set for each\
+        committee $i$ derived from epoch $e_{j - 3}$ view
+        \State Set lottery thresholds $\{(T^{\text{lead}, i}, T^{\text{sync}, i})\}_{i=1}^K$ for each\
+        committee $i$ derived from epoch $e_{j - 1}$ view
+        \State Set ${\eta_j \leftarrow \mathcal{F}_{\text{LB}}(e_j, C_{\text{loc}})}$
+
+        \State Parse $B$ as ${(h, sl, \textsf{state}, v^{\text{vrf}}, \pi^{\text{sl}}, \sigma_{\text{KES}}^{\text{sl}})}$
```

Comment:
What it `state`?

Reply:
How can ledger state be parsed from a block?

Reply:
Meaning the state of the local leader chain with this block on top of it. This state must also be signed with KES, I will add a block creation procedure with all minor fixes to make it clear

Reply:
Probably you mean state root hash?